third_party/move/move-compiler/src/cfgir/translate.rs

```diff
@@ -423,7 +432,33 @@ fn function_body(
 
             let (mut cfg, infinite_loop_starts, diags) =
                 BlockCFG::new(start, &mut blocks, &block_info);
+
             context.env.add_diags(diags);
+            if context.env.flags().debug() {
```

Comment:
That is unusual. The idiomatic way would be to have this compile time protected, as in

```
#[cfg(debug_assertions)]
eprintln!(...
```

But understood that seems to be a result of the env vars you have introduced before for printing error details. 

We should discuss whether we should really use this like this. (Not as part of this PR.) There are crates for logging which might be better suited than an explicit println.

Reply:
I'll remove the extra printing before submitting,

Reply:
This kind of thing is really not for typical use, just once the target source code is identified and you want to see everything the compiler is doing with it.  That's why it's an awkward environment variable, MOVE_COMPILER_DEBUG.

I changed my mind, probably won't remove this one, as seeing the state between compiler stages can be very useful if we have more debugging to do there.  I wonder if I can get the compiled function name easily enough to have this be a string to match instead, though.

Reply:
SG, as said this is not for this PR. We should have a general logging model using the [log crate similar as in the prover](https://github.com/aptos-labs/aptos-core/blob/55398e79f7f231dccf6f25cf01a648315193ede4/third_party/move/move-prover/boogie-backend/src/boogie_wrapper.rs#L118C59-L118C59) for this kind of thing, at least in v2. Not caring that much what we do in v1.

Reply:
ok.